src/drivers/dw/dma.c

```diff
@@ -546,14 +546,17 @@ static int dw_dma_set_config(struct dma_chan_data *channel,
 
 		channel->desc_count = config->elem_array.count;
 
-		/* allocate descriptors for channel */
+		/*
+		 * Allocate descriptors for channel. They must be cache-line
+		 * size aligned to avoid corrupting adjacent memory when
+		 * synchronizing caches.
+		 */
 		if (dw_chan->lli)
 			rfree(dw_chan->lli);
 
-		dw_chan->lli = rmalloc(SOF_MEM_ZONE_SYS_RUNTIME, 0,
-				       SOF_MEM_CAPS_RAM | SOF_MEM_CAPS_DMA,
-				       sizeof(struct dw_lli) *
-				       channel->desc_count);
+		dw_chan->lli = rballoc_align(0, SOF_MEM_CAPS_RAM | SOF_MEM_CAPS_DMA,
```

Comment:
Per verbal discussion: I think where we really want to fix this is in the Zephyr wrapper implementation of r*alloc*(): if we get an allocation with the SOF_MEM_CAPS_DMA flag set, then we should be automatically requesting cache-line alignment and padding the requested size to an integer number of cache lines (so the heap doesn't try to place other data in the incoherent suffix that DMA will clobber).

Reply:
@andyross @lgirdwood alright, so do we drop this one while I make the suggested fix?

Reply:
No lets add this in the comments and commit message and apply. This does fix something, so is needed -  it can be updated when @andyross zephyr fix is upstream.

Reply:
@lgirdwood ok, I'll add comments, although I didn't understand which fixes from @andyross you mean - IIUC he was proposing changes in zephyr/wrapper.c i.e. in our code, not in Zephyr itself.

Reply:
...although, I must say, I'm also not quite convinced, that implementing that fix in a wrapper would be correct, since it also should be applied for other DMA allocations on non-cache coherent systems, e.g. for any other DMA allocations on Xtensa

Reply:
Right, my suggestion was to do it in SOF code, but at the level where you know it's a DMA allocation (from the flags argument to rmalloc).   That seems like the right spot to me.  The Zephyr heap code is just a block of memory, it doesn't understand the DMA controller architecture.  But it's true, the wrapper would have to have some level of architecture-intelligence: not all SOF platforms are going to share the same characteristics, etc...